document_merge_service/api/management/commands/upload_local_templates.py

```diff
@@ -1,35 +1,66 @@
+import glob
 import os
 
-from django.conf import settings
 from django.core.files.storage import DefaultStorage
 from django.core.management.base import BaseCommand
 
 from document_merge_service.api.models import Template
 
 
 class Command(BaseCommand):
-    help = "Add filesystem files to configured storage backend"
+    help = "Upload local template files to configured storage backend"
 
     def add_arguments(self, parser):
-        parser.add_argument("--dry-run", dest="dry", action="store_true", default=False)
+        parser.add_argument(
+            "-s",
+            "--source",
+            help="Glob-style path to the template files that should be uploaded. E.g. `/tmp/templates/*.docx`",
+            dest="source",
+            type=str,
+            required=True,
+        )
+        parser.add_argument(
+            "--dry-run",
+            help="Only show what files would be uploaded to the storage backend; don't actually upload them.",
+            dest="dry",
+            action="store_true",
+            default=False,
+        )
 
     def handle(self, *args, **options):
         storage = DefaultStorage()
-        for template in Template.objects.all():
+
+        for path in glob.iglob(options["source"]):
+            filename = os.path.basename(path)
+
             try:
-                with open(
-                    os.path.join(settings.MEDIA_ROOT, template.template.name), "rb"
-                ) as file:
-                    if not options.get("dry"):
+                template = Template.objects.get(template=filename)
+            except Template.DoesNotExist:
+                self.stdout.write(
+                    self.style.WARNING(f'No template for filename "{filename}" found')
+                )
+                continue
+
+            if not options.get("dry"):
+                try:
+                    with open(path, "rb") as file:
+                        storage.delete(template.template.name)
```

Comment:
django-storages offeres a file overwrite option for some backend, so the delete might not always be necessary.
> AWS_S3_FILE_OVERWRITE (optional: default is True)
    By default files with the same name will overwrite each other. Set this to False to have extra characters appended.

Reply:
Yes, but this command is supposed to work with all backends. It must be a backend-independent solution.